.agents/skills/commit/SKILL.md

```
@@ -237,15 +237,17 @@ If generated files are part of a larger change (e.g., you added a new GraphQL fi
 
 ## Upgrade Notes
 
-When changes include upgrade note files (typically in `upgrade/` directory or files matching `*UPGRADE*` patterns), always group them into their own separate commit.
+When changes include upgrade note files (typically in `upgrade/` directory or files matching `*UPGRADE*` patterns), group them into their own separate commit by default.
+
+If the user explicitly asks for a single commit, one commit, or to include all changes in the same commit, obey that request and include upgrade notes in the same commit as the related changes.
 
 **Commit message:** `added upgrade notes` — no body needed.
 
 ```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)